org.eclipse.gef/src/org/eclipse/gef/ui/parts/TreeViewer.java

```diff
@@ -223,31 +234,22 @@ public void widgetDefaultSelected(SelectionEvent e) {
 	 */
 	@Override
 	public void reveal(EditPart part) {
-		if (!(part instanceof TreeEditPart treePart))
+		if (!(part instanceof TreeEditPart treePart)) {
 			return;
-		Tree tree = (Tree) getControl();
-		Widget widget = treePart.getWidget();
-		if (widget instanceof TreeItem)
-			tree.showItem((TreeItem) widget);
+		}
+		if (treePart.getWidget() instanceof TreeItem treeItem) {
+			getControl().showItem(treeItem);
+		}
 	}
 
 	private void showSelectionInTree() {
-		if (ignore || getControl() == null || getControl().isDisposed())
+		if (ignore || getControl() == null || getControl().isDisposed()) {
 			return;
-		List selection = getSelectedEditParts();
-		Tree tree = (Tree) getControl();
-		List treeParts = new ArrayList();
-		for (int i = 0; i < selection.size(); i++) {
-			TreeEditPart part = (TreeEditPart) selection.get(i);
-			if (part.getWidget() instanceof TreeItem)
-				treeParts.add(part);
-		}
-		TreeItem[] treeItems = new TreeItem[treeParts.size()];
-		for (int i = 0; i < treeParts.size(); i++) {
-			TreeEditPart part = (TreeEditPart) treeParts.get(i);
-			treeItems[i] = (TreeItem) part.getWidget();
 		}
-		tree.setSelection(treeItems);
+		TreeItem[] treeItems = getSelectedEditParts().stream().map(TreeEditPart.class::cast)
+				.filter(p -> p.getWidget() instanceof TreeItem).map(TreeEditPart::getWidget).map(TreeItem.class::cast)
+				.toArray(TreeItem[]::new);
+		getControl().setSelection(treeItems);
```

Comment:
I'm not sure if this is really more readable... Perhaps splitting up the method calls into more lines makes it cleaner?

Reply:
I'm now rather used to the stream style but I can understand the problems with this here. Let me try reformating it.

Reply:
Hmh it looks like Eclipse's format is condensing it and I didn't find any option on controlling formatting of stream. 

The reason I changed is is that it is more efficient and for me more maintainable then the previous code.  But I'm happy to change it back into something with loops if you think I overdid it.

Reply:
> Hmh it looks like Eclipse's format is condensing it and I didn't find any option on controlling formatting of stream.

You can turn off the formatter by using these special comments (which needs to be enabled in the formatter profile):

> // @formatter:off
> ...
> // @formatter:on

Alternatively, you can just add empty comments at the end of a line:

> getSelectedEditParts() //
> 		.stream() //
>		.map(TreeEditPart.class::cast) //
>		.filter(p -> p.getWidget() instanceof TreeItem) //
> ...

> The reason I changed is is that it is more efficient and for me more maintainable then the previous code. But I'm happy to change it back into something with loops if you think I overdid it.

I'm fine with using Streams. It's just the formatting that's a little off-putting :sweat_smile:

Reply:
I know the formatter on/off thingy but it seems we have it not yet enabled. So I used the comment approach which I didn't know existed.